File: Java/SloanE6/Enigma.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package sloan.e7.enigma;

import java.util.Random;

/**
 *
 * @author O
 */
public class Enigma {
    
    String message;
    String codedMessage;
    int key;
    
    public Enigma(){
        message = "Default Message";
        codedMessage = "DEFAULT";
        key = 1;
    }
    
    public void setMessage(String s, int k)
    { 
        message = s;
        key = k;
        if (key > 50 || key < 0)
        {
            key = 1;
        }
        encode();
       
    }
    
    public void setMessage(String s)
    {
        
        message = s;
        Random random = new Random();
        random.setSeed(System.currentTimeMillis());
        key = random.nextInt(50) + 1;
        encode();
        
    }
    
    public void setCodedMessage(String encodedMsg, int k)
    {
        codedMessage = encodedMsg;
        key = k;
        decode();
        
    }
    
    public String getCodedMessage()
    {
        return codedMessage;
        
    }
    
    public String getDecodedMessage()
    {
        return message;
    }
    
    public int getKey()
    {
        return key;
    }
    
    private void encode()
    {
        String encodedMessage = "";
    
        for(int i = 0; i < message.length(); ++i)
        {
            char l = message.charAt(i);
            int addKey = l + key;
            if (addKey > 126)
            {
                addKey -= 127;
                addKey += 32;
                
            }
            char m = (char)addKey;
            encodedMessage += m;
        }
        codedMessage = encodedMessage;
    }
    
    private void decode()
    {
        String decodedMessage = "";
        for(int j = 0; j < codedMessage.length(); ++j)
        {
            char n = codedMessage.charAt(j);
            int subKey = n - key;
            if(subKey < 32)
            {
                subKey += 127;
                subKey -= 32;
            }
            char o = (char)subKey;
            decodedMessage += o;
        }
        message = decodedMessage;
    }
}

File: C++/C++ II/SloanP8/Enigma.cpp
#include "stdafx.h"
#include "Enigma.h"
#include <string>
using namespace std;

void Enigma::Encode(string message, int key)
{
	string encodedMsg = "";
	for(unsigned int i = 0; i < message.size(); ++i)
	{
		char msg = message[i];
		int addKey = msg + key;
		if(addKey > 126)
		{
			addKey -= 126;
			addKey += 32;
		}
		char conv = static_cast<char>(addKey);
		encodedMsg += conv;
	}
	codedMessage = encodedMsg;

}
void Enigma::Decode(string cMsg, int k)
{
	string decodedMessage = "";
	codedMessage = cMsg;	
	key = k;
	for (unsigned int j = 0; j < codedMessage.size(); ++j)
	{
		char l = codedMessage[j];
		int subKey = l - key;
		if(subKey < 32)
		{
			subKey += 126;
			subKey -= 32;
		}
		char conv = (char)subKey;
		decodedMessage += conv;
	}
	message = decodedMessage;
}

void Enigma::setMessage(string msg, int k)
{
	message = msg;
	key = k;
	Encode(message,k);
	
}

void Enigma::setCodedMessage(string cMsg, int k)
{
	codedMessage = cMsg;
	key = k;
	Decode(codedMessage,k);
	
}

void Enigma::setKey(int k)
{
	key = k;
}